presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -987,6 +1021,13 @@ private boolean canQueueMore()
         }
     }
 
+    private int getActiveWorkerCount()
+    {
+        AllNodes allNodes = nodeManager.getAllNodes();
+        Set<InternalNode> activeWorkers = Sets.difference(allNodes.getActiveNodes(), allNodes.getActiveResourceManagers());
```

Comment:
need to exclude `activeCatalogServers` from this as well to make it work for all setup.